internal/errors/error.go

```diff
@@ -117,16 +117,15 @@ func Convert(e error) *Err {
 	if e == nil {
 		return nil
 	}
-	var err *Err
-	if As(e, &err) {
+	if err, ok := e.(*Err); ok {
```

Comment:
I went back to doing this as a direct cast instead of an `As` because many places in the code do a `fmt.Errorf("context %w`.  The convert function then losses that context when wrapping the error higher up.  Once all code paths have been converted to domain errors this problem will go away

Reply:
Would you mind adding a comment describing when the TODO can be accomplished or why it isn't done now?  Something like "When all code paths have been converted to domain errors change this to using As instead of casting the error."